junit5/src/main/java/cz/xtf/junit5/extensions/SkipForCondition.java

```diff
@@ -32,8 +33,11 @@ public ConditionEvaluationResult evaluateExecutionCondition(ExtensionContext con
 
 	public static ConditionEvaluationResult resolve(SkipFor skipFor) {
 		Image image = Image.resolve(skipFor.image());
+		DockerImageMetadata metadata = DockerImageMetadata.get(OpenShifts.master(), image);
+		String[] imageNameParts = metadata.labels().get("name").split("/");
+		String actualImageName = imageNameParts.length == 2 ? imageNameParts[1] : imageNameParts[0];
 		Pattern name = Pattern.compile(skipFor.name());
-		if (name.matcher(image.getRepo()).matches()) {
+		if (name.matcher(actualImageName).matches()) {
```

Comment:
I believe this is backwards incompatible, right? We're still allowing regexes, but now you're checking against the `name` label instead of checking against the image name. All consumers of `@SkipFor` would have to change the `name` in the annotations now.

I propose to either:
* Deprecate SkipFor and create a new annotation which will check against metadata (or will allow both metadata and image name checks).
* Make the `name` parameter in `@SkipFor` optional, and add another parameter for metadata parsing. Only one can be defined at a time and the implementation will check against what's defined.

Reply:
Yes, you are right. I went with the second option. You can check it out. I will let you know once I will finish test.

Reply:
The backward compatibility might be guaranteed also by matching `name` from `@SkipFor` against both:
-  image.get()
- actualImageName

If any of the them matches then test is skipped and avoid adding new property into `@SkipFor` annotation, wdyt?

Reply:
Yeah. It changes old functionality, however keep backward compatibility to some level. Strictly speaking, it may results to skipping tests that should not be skipped. However I don't see this is a real scenario. Image name is very similar to docker name label.

Reply:
I'm not sure if I like it. I think it's a shady way of changing the behavior. When new people start writing tests, they'll see that in most existing tests the regexp matches `image.getRepo()`. They'll likely go on thinking that's the expected behavior and won't know that it also checks the label. Yes, it's documented, but people don't read documentation that often.

Additionally, there may be differences between the two, and adding this behavior may break existing tests. It's unlikely but it can happen. I'd still recommend one of my suggested solutions. Let's see what the others think.